src/main/java/sdk/sample/common/CommonSdk.java

```diff
@@ -0,0 +1,321 @@
+     */
+    public static <T> void waitForNoANFResource(NetAppManagementClient anfClient, String resourceId, int intervalInSec, int retries, Class<T> clazz)
+    {
+        for (int i = 0; i < retries; i++)
+        {
+            Utils.threadSleep(intervalInSec * 1000);
+
+            try
+            {
+                switch (clazz.getSimpleName())
+                {
+                    case "NetAppAccountInner":
+                        NetAppAccountInner account = anfClient.getAccounts().getByResourceGroup(ResourceUriUtils.getResourceGroup(resourceId),
+                                ResourceUriUtils.getAnfAccount(resourceId));
+                        if (account == null)
+                            return;
+
+                        continue;
+
+                    case "CapacityPoolInner":
+                        CapacityPoolInner pool = anfClient.getPools().get(ResourceUriUtils.getResourceGroup(resourceId),
+                                ResourceUriUtils.getAnfAccount(resourceId),
+                                ResourceUriUtils.getAnfCapacityPool(resourceId));
+                        if (pool == null)
+                            return;
+
+                        continue;
+
+                    case "VolumeInner":
+                        VolumeInner volume = anfClient.getVolumes().get(ResourceUriUtils.getResourceGroup(resourceId),
+                                ResourceUriUtils.getAnfAccount(resourceId),
+                                ResourceUriUtils.getAnfCapacityPool(resourceId),
+                                ResourceUriUtils.getAnfVolume(resourceId));
+                        if (volume == null)
+                            return;
+
+                        continue;
+
+                    case "SnapshotInner":
+                        SnapshotInner snapshot = anfClient.getSnapshots().get(ResourceUriUtils.getResourceGroup(resourceId),
+                                ResourceUriUtils.getAnfAccount(resourceId),
+                                ResourceUriUtils.getAnfCapacityPool(resourceId),
+                                ResourceUriUtils.getAnfVolume(resourceId),
+                                ResourceUriUtils.getAnfSnapshot(resourceId));
+                        if (snapshot == null)
+                            return;
+                }
+            }
+            catch (Exception e)
+            {
+                Utils.writeWarningMessage(e.getMessage());
+                break;
+            }
+        }
+    }
+
+    public static void waitForVolumeSuccess(NetAppManagementClient anfClient, String resourceGroupName, String accountName, String poolName, String volumeName)
+    {
+        waitForVolumeSuccess(anfClient, resourceGroupName, accountName, poolName, volumeName, 10, 60);
+    }
+
+    public static void waitForVolumeSuccess(NetAppManagementClient anfClient, String resourceGroupName, String accountName, String poolName, String volumeName, int intervalInSec, int retries)
```

Comment:
I would say to be safer, lets get the old function back "waitForAnfResource" where it's more generic and check the success status for any ANF resource. and this function needs to be called after the creation of each resource

Reply:
But that one does not check on the status of the volume

Reply:
I'll add it back but in my opinion that is overengineering for a code example and over-generic.
Also, if you look at the example as it was before then it is only calling waitForAnfResource after volume creation. So I will keep it like that.

Reply:
@hildurhodd The function "waitForVolumeSuccess" is checking both volume status at the same time which one might be failing. My whole point is to have the already implemented function "waitForANFResource" after each resource creation, if the resource is not done or failing for any reasons, it should not continue.